File: Crm.Article/Model/Enums/DiscountType.cs
namespace Crm.Article.Model.Enums
{
	public enum DiscountType
	{
		NoDiscount = 0,
		Percentage = 1,
		Absolute = 2
	}

	public static class DiscountTypeExtensions
	{
		public static string GetSuffix(this DiscountType dt)
		{
			switch (dt)
			{
				case DiscountType.Percentage:
					return "%";
				default:
					return "";
			}
		}
	}
}

File: Crm.Article/Content/js/DiscountType.js
(function() {
	// this will be replaced by odata $metadata
	if (!window.Crm) {
		window.Crm = {};
	}
	if (!window.Crm.Article) {
		window.Crm.Article = {};
	}
	if (!window.Crm.Article.Model) {
		window.Crm.Article.Model = {};
	}
	if (!window.Crm.Article.Model.Enums) {
		window.Crm.Article.Model.Enums = {};
	}
	if (!window.Crm.Article.Model.Enums.DiscountType) {
		window.Crm.Article.Model.Enums.DiscountType = {
			NoDiscount: 0,
			Percentage: 1,
			Absolute: 2
		};
	}
})();